pkg/object/s3.go

```diff
@@ -306,39 +323,46 @@ func newS3(endpoint, accessKey, secretKey string) (ObjectStorage, error) {
 		ep         string
 	)
 
-	if len(hostParts) == 1 { // take endpoint as bucketname
+	if len(hostParts) == 1 {
+		// take endpoint as bucketname
 		bucketName = hostParts[0]
 		if region, err = autoS3Region(bucketName, accessKey, secretKey); err != nil {
 			return nil, fmt.Errorf("Can't guess your region for bucket %s: %s", bucketName, err)
 		}
-	} else { // get region in endpoint
-		if strings.Contains(uri.Host, ".amazonaws.com") {
-			// standard s3
-			// [BUCKET].s3-[REGION].[REST_OF_ENDPOINT]
-			// [BUCKET].s3.[REGION].amazonaws.com[.cn]
-			hostParts = strings.SplitN(uri.Host, ".s3", 2)
-			bucketName = hostParts[0]
-			endpoint = "s3" + hostParts[1]
-			if strings.HasPrefix(endpoint, "s3-") || strings.HasPrefix(endpoint, "s3.") {
-				endpoint = endpoint[3:]
-			}
-			if strings.HasPrefix(endpoint, "dualstack") {
-				endpoint = endpoint[len("dualstack."):]
-			}
-			if endpoint == "amazonaws.com" {
-				endpoint = awsDefaultRegion + "." + endpoint
-			}
-			region = strings.Split(endpoint, ".")[0]
-			if region == "external-1" {
+	} else {
+		// get region or endpoint
+		if uri.Path != "" {
```

Comment:
What will happen if the path is `/` ?

Reply:
We can trim `/` before check to prevent this situation